components/net/protocols/mod.rs

```diff
@@ -128,3 +141,20 @@ pub fn get_range_request_bounds(range: Option<Range>, len: u64) -> RangeRequestB
 pub fn partial_content(response: &mut Response) {
     response.status = StatusCode::PARTIAL_CONTENT.into();
 }
+
+pub trait ProtocolUrlExt {
```

Comment:
I think a helper function in `components/net/fetch/methods.rs` is better. A trait indicate it needs to extend to more types in the future, while we just need `is_potentially_trustworthy_with_custom_protocols` for internal usage.

Reply:
I think maybe adding this to `ProtocolRegistry` might be a better choice? I'll get some food and get back to this

Reply:
Add to ProtocolRegistry is great too